internal/cmd/all/list.go

```diff
@@ -0,0 +1,157 @@
+package all
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/hetznercloud/cli/internal/cmd/base"
+	"github.com/hetznercloud/cli/internal/cmd/certificate"
+	"github.com/hetznercloud/cli/internal/cmd/firewall"
+	"github.com/hetznercloud/cli/internal/cmd/floatingip"
+	"github.com/hetznercloud/cli/internal/cmd/image"
+	"github.com/hetznercloud/cli/internal/cmd/iso"
+	"github.com/hetznercloud/cli/internal/cmd/loadbalancer"
+	"github.com/hetznercloud/cli/internal/cmd/network"
+	"github.com/hetznercloud/cli/internal/cmd/output"
+	"github.com/hetznercloud/cli/internal/cmd/placementgroup"
+	"github.com/hetznercloud/cli/internal/cmd/primaryip"
+	"github.com/hetznercloud/cli/internal/cmd/server"
+	"github.com/hetznercloud/cli/internal/cmd/sshkey"
+	"github.com/hetznercloud/cli/internal/cmd/volume"
+	"github.com/hetznercloud/cli/internal/hcapi2"
+	"github.com/hetznercloud/cli/internal/state"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+var listCmd = base.Cmd{
+	BaseCobraCommand: func(client hcapi2.Client) *cobra.Command {
+		cmd := &cobra.Command{
+			Use:   "list FLAGS",
+			Short: "List all resources",
+		}
+
+		cmd.Flags().StringP("selector", "l", "", "Selector to filter by labels")
+
+		cmd.Flags().Bool("paid", false, "Only list paid resources")
```

Comment:
Maybe this is easier to understand?

```suggestion
		cmd.Flags().Bool("paid", false, "Only list resources that cost money")
```

Reply:
I think introducing the notion of paid with this flag might lead to confusions.

I would rather use another flag name (e.g --owned) or invert it with --include-read-only-resource (yeah this is too long, but maybe you folks have an idea?). Using a --all flag might be a bit repetitive `all list --all`.

Reply:
What would be the usecase to list all static resources? I think the user would rather like to see their resources.

---

I think we are conflating two discussions:

- _Paid_ (servers, IPs, LBs, Volumes) vs. _free_ (Network, SSH Key, Placement Group)
- _Project Resouces_ (Server, LB, PG, Snapshots+Backups Images) vs _Static/Public_ (System+App Images, Public ISOs, Location, Network Zone, Server Types, Load Balancer Types)

Reply:
We have a similar pattern in `console.hetzner.cloud` on the Project Card View. Here it shows all paid resources as icons for every project and unpaid resources are grouped under "Other":

![screenshot-paid-unpaid](https://github.com/hetznercloud/cli/assets/3393079/c5b522cd-d29c-44f0-a37f-90fc5f197b41)

Reply:
Oh alright, so list all is only for project resources ? In that case, the paid flag name make more sense. But this isn't This wasn't clear from the current description of the command, I would have expected to list all resource (+ static/public)

Reply:
The long description now explains the command in more detail and explicitly states that static and public resources are not included.